### PR TITLE
Update go version to 1.17

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,6 +69,167 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
```
Comment (Skarlso) [inline go.mod:72]: result of `go mod tidy`.

**Review by palemtnrider — APPROVED** [review]
LGTM

**Comment by palemtnrider** [comment]
This PR is missing a few additional changes.  I'll add a commit with the additional changes.

**Comment by palemtnrider** [comment]
@Skarlso when you have a chance, please take a look at the additional commits I added and see if there is anything else we should update for the move to 1.17.

**Comment by Skarlso** [comment]
@palemtnrider Oh, excellent cool! Thanks, I'll look them over.

**Comment by Skarlso** [comment]
Changes look good to me! Well done. :)

**Review by josecordaz — APPROVED** [review]

**Pull request merged**